core/src/main/java/org/sonarsource/sonarlint/core/client/api/common/InputFileEdit.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * SonarLint Core - Implementation
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.client.api.common;
+
+import java.util.List;
+import org.sonar.api.batch.fs.InputFile;
+
+public interface InputFileEdit {
```

Comment:
This one could be `ClientInputFileEdit`, and we should return a `ClientInputFile` from `target()`. With the current SLI implementation it's not a problem because we don't check the file (we assume all the quick fixes target the analyzed file) but we might want to check this later